ui/app/AppLayouts/Wallet/views/walletconnect/WalletConnectSDK.qml

```diff
@@ -1,326 +1,338 @@
-            const jsCode = "wcResult = {}; try { const callRes = wc." + methodName + "(" + (paramsStr ? (paramsStr) : "") + "); wcResult = {state: " + successState + ", error: null, result: callRes}; } catch (e) { wcResult = {state: " + errorState + ", error: \"Exception: \" + e.message};  }; wcResult"
-            root.runJavaScript(jsCode,
-                function(result) {
-                    timer.restart()
-                }
-            )
+        function onSessionExpire(details)
+        {
+            console.debug(`@dd TODO WalletConnectSDK.onSessionExpire; details: ${JSON.stringify(details, null, 2)}`)
         }
 
-        function startListeningForEvents() {
-            const jsCode = "
-                try {
-                    function processWCEvents() {
-                        window.wcEventResult = [];
-                        window.wcEventError = null
-                        window.wc.registerForSessionRequest((event) => {
-                            window.wcEventResult.push({name: 'session_request', payload: event});
-                        });
-                    }
-                    processWCEvents();
-                } catch (e) {
-                    window.wcEventError = e
-                }
-                window.wcEventError"
+        function onSessionRequest(details)
+        {
+            console.debug(`@dd WalletConnectSDK.onSessionRequest; details: ${JSON.stringify(details, null, 2)}`)
+            root.sessionRequestEvent(details)
+        }
 
-            root.runJavaScript(jsCode,
-                function(result) {
-                    if (result) {
-                        console.error("startListeningForEvents: processWCEvents error", result)
-                        return
-                    }
-                }
-            )
-            eventsTimer.start()
+        function onSessionRequestSent(details)
+        {
+            console.debug(`@dd TODO WalletConnectSDK.onSessionRequestSent; details: ${JSON.stringify(details, null, 2)}`)
         }
 
-        function init(projectId) {
-            d.requestSdkAsync(generateSdkCall("init", `"${projectId}"`, RequestCodes.SdkInitSuccess, RequestCodes.SdkInitError))
+        function onSessionEvent(details)
+        {
+            console.debug(`@dd TODO WalletConnectSDK.onSessionEvent; details: ${JSON.stringify(details, null, 2)}`)
         }
 
-        function getPairings(projectId) {
-            d.requestSdk("getPairings", `null`, RequestCodes.GetPairings, RequestCodes.GetPairingsError)
+        function onProposalExpire(details)
+        {
+            console.debug(`@dd WalletConnectSDK.onProposalExpire; details: ${JSON.stringify(details, null, 2)}`)
+            root.pairSessionProposalExpired()
         }
     }
 
     ListModel {
         id: pairings
     }
-}
\ No newline at end of file
+
+    WebChannel {
+        id: statusChannel
+        registeredObjects: [statusObject]
+    }
+
+    WebEngineView {
```

Comment:
I was trying to avoid using `WebEngine` so we can remove it soon. Does `WebChannel` work with QML `WebView`, or is the presence of qt's browser a vital requirement?

Reply:
As far as I see, there is not option to use WebChannel in WebView, that's why I had to switch to WebEngineView.